src/main/java/com/dataguise/saas/service/ProtectionTaskService.java

```diff
@@ -0,0 +1,611 @@
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+@Service
+public class ProtectionTaskService {
+
+    private static Logger logger = Logger.getLogger(ProtectionTaskService.class);
+
+    @Autowired
+    private DgSecureRestDriver restDriver;
+
+    @Autowired
+    private SettingsController settingsController;
+
+    @Autowired
+    private TaskBuilderService taskBuilderService;
+
+    /**
+     * Create a Protection task
+     * @param protectionTask
+     * @return
+     * @throws BadGatewayException
+     */
+    public Integer createTask(@RequestBody DgMaskerTaskDTO protectionTask, String sourceModule) throws BadGatewayException {
+        Integer taskId = null;
+        try {
+            taskId = restDriver.createMaskerTask(protectionTask, sourceModule);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Protection task created with id: " + taskId);
+        return taskId;
+    }
+
+    /**
+     * Edit a protection task
+     * @param id
+     * @param protectionTask
+     * @return
+     * @throws BadGatewayException
+     * @throws BadRequestException
+     */
+    public Integer editTask(@PathVariable Integer id, @RequestBody DgMaskerTaskDTO protectionTask)
+            throws BadGatewayException, BadRequestException {
+        Integer taskId = null;
+        if( id == null) {
+            throw new BadRequestException("Task " + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage());
+        }
+        try {
+            taskId = restDriver.updateMaskerTask(protectionTask, id);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Protection task edited with id: " + taskId);
+        return taskId;
+    }
+
+    /**
+     * Execute a protection task.
+     * @param taskId
+     * @return
+     * @throws BadRequestException
+     * @throws BadGatewayException
+     */
+    public Integer executeTask(@PathVariable Integer taskId) throws BadRequestException, BadGatewayException {
+        if( taskId == null) {
+            throw new BadRequestException("Task " + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage());
+        }
+        Integer taskInstanceId = null;
+        try {
+            taskInstanceId = restDriver.executeMaskerTask(taskId);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Protection task executed with instance id: " + taskInstanceId);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/ProtectionTaskService.java L101
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a406025194>*